### ISSUE TITLE
Entity is detached from DAO context exception, when only using getters

### ISSUE THREAD

**Issue by Samzerge** [greenrobot/greenDAO#473]
I'm getting this exception when retrieving objects from database.

`List<Point> listPoints = AppController.daoSession.getPointDao().queryBuilder().where(PointDao.Properties.CoordID.eq(500L)).list();`
`Point selectedPoint = listPoints.get(0);`
`PointType pointType = selectedPoint.getPointType();`
`List<ProductPoint> productPointList = pointType.getProductPoints();`

The exception is thrown at this method:     **pointType.getProductPoints()**
I read that this happened when creating an instance outside of the database but in my case this happens when only using getters. I have checked the database and the data is inserted correctly.
Does someone knows why is this happening?

**Comment by greenrobot-team** [comment]
In general, have a look at [how relations work and are resolved in greenDAO](http://greenrobot.org/greendao/documentation/relations/).

Without an example project I can only guess: make sure to insert the referred to entities (`PointType` and `ProductPoint`) before adding a reference to them in your `Point` and `PointType` entities.

So, first store `ProductPoint`, then `PointType` with references to `ProductPoint`, then `Point` with references to `PointType`.

Also as per [this comment](https://github.com/greenrobot/greenDAO/issues/142#issuecomment-117293272), make sure the `PointType` entity is attached to a session. If not, attach one by calling `__setDaoSession(DaoSession daoSession)`.
-ut

**Comment by Samzerge** [comment]
Well, I solved the problem. I'm not sure if this the optimal solution but it works. I first noticed that this exception ocurred only if I tried to call the method just after the data had been inserted to the database. If I worked with a database that already existed it didn't throw the exception. So I cleared the daoSession and started it again after the initial insertion.

`daoSession.clear();`
`daoSession = AppController.daoMaster.newSession();`
